src/components/__tests__/ProgressCircle.spec.ts

```diff
@@ -67,91 +67,127 @@ describe("ProgressCircle", () => {
+        expect(progress.state().alertMessage).toEqual("");
     });
 
-    it("renders a circle with the text set to Invalid when the maximum value is less than 1", () => {
+    it("destroys progress circle on unmount", () => {
+        spyOn(progressbar.Circle.prototype, "destroy").and.callThrough();
+        const destroy = progressbar.Circle.prototype.destroy as jasmine.Spy;
         spyOnCircle();
 
-        const progress = newCircleInstance({
-            animate: false,
-            maximumValue: -1,
-            showPercentage: true,
-            value: 80
-        });
+        const progress = newCircleInstance({ value: 80 });
         progress.componentDidMount();
+        progress.componentWillUnmount();
 
-        expect(progressCircle.text.textContent).toBe("Invalid");
+        expect(destroy).toHaveBeenCalled();
     });
 
-    it("renders a circle with negative values when the value is less than zero", () => {
-        spyOnCircle();
+    describe("renders a circle with ", () => {
+        it("no text when display text is none", () => {
+            spyOnCircle();
 
-        const progress = newCircleInstance({ animate: false, value: -200 });
-        progress.componentDidMount();
+            const progress = newCircleInstance({ value: 50, text: "none" });
+            progress.componentDidMount();
 
-        expect(progressCircle.text.textContent).toBe("-200%");
-    });
+            expect(progressCircle.text.textContent).toBe("");
+        });
 
-    it("renders a circle without percentage", () => {
-        spyOnCircle();
+        it("only the value when display text is value", () => {
+            spyOnCircle();
 
-        const progress = newCircleInstance({ value: 10, showPercentage: false });
-        progress.componentDidMount();
+            const progress = newCircleInstance({ value: 10, text: "value" });
+            progress.componentDidMount();
 
-        expect(progressCircle.text.textContent).not.toContain("%");
-    });
+            expect(progressCircle.text.textContent).not.toContain("%");
+        });
 
-    it("renders a circle with text greater than 100% when the value is greater than the maximum", () => {
-        spyOnCircle();
+        it("the text -- when no value is specified", () => {
+            spyOnCircle();
 
-        const progress = newCircleInstance({ value: 180 });
-        progress.componentDidMount();
+            const progress = newCircleInstance({ value: undefined });
+            progress.componentDidMount();
 
-        expect(progressCircle.text.textContent).toBe("180%");
-        expect(progressCircle.animate).toHaveBeenCalledWith(1);
-    });
+            expect(progressCircle.text.textContent).toBe("--");
+        });
 
-    it("has the class widget-progress-circle-alert when the maximum value is less than one", () => {
-        const progress = renderProgressCircle({ value: 20, maximumValue: 0 });
+        it("the text set to Invalid when the maximum value is less than 1", () => {
```

Comment:
`text is set to invalid when the maximum value is less than 1`

Reply:
The test sentence changes if this is changed.
Currently, it reads....ProgressCircle renders a circle with the text set to Invalid when the maximum value is less than 1